hibernate-core/src/main/java/org/hibernate/cfg/Environment.java

```diff
@@ -177,28 +178,31 @@ public static void verifyProperties(Map<?,?> configurationValues) {
 		//Set USE_REFLECTION_OPTIMIZER to false to fix HHH-227
 		GLOBAL_PROPERTIES.setProperty( USE_REFLECTION_OPTIMIZER, Boolean.FALSE.toString() );
 
-		try {
-			InputStream stream = ConfigHelper.getResourceAsStream( "/hibernate.properties" );
+		LinkedList<ClassLoader> classLoaders = new LinkedList<>();
```

Comment:
Please use an `ArrayList` or `Arrays.asList( Thread.currentThread().getContextClassLoader(), Environment.class.getClassLoader() )` as `LinkedList` is rarely appropriate.

Reply:
It is in this case I think, since I want to make sure to respect the order in which ClassLoaders are interrogated, as before this change.

Reply:
List is inherently ordered

Reply:
Even an array would work here as the number of items is known and constant

Reply:
> List is inherently ordered

I seem to still confuse Collection.iterator() with List.iterator() ... after all these years, it's just  .... sad :-(((